CoordImporter/Parser/Parsers.cs

```diff
@@ -1,113 +1,137 @@
-﻿using CoordImporter.Managers;
-using CSharpFunctionalExtensions;
-using Dalamud.Game.Text;
-using System;
+﻿using System;
 using System.Globalization;
 using System.Linq;
 using System.Numerics;
 using System.Text.RegularExpressions;
+using CoordImporter.Managers;
+using CSharpFunctionalExtensions;
+using Dalamud.Game.Text;
+using Dalamud.Plugin.Services;
 
 namespace CoordImporter.Parser;
 
-public record struct MapData(
-    uint TerritoryId,
-    uint RowId
-);
+public interface ITrackerParser
+{
+    public bool CanParseLine(string inputLine);
+
+    public Result<MarkData, string> Parse(string inputLine);
+}
 
-internal static class ParsingExtensions
+public abstract class Parsers : ITrackerParser
```

Comment:
ah can we call the class `Parser` singular? since it's one parser? i named the file `Parsers.cs` because multiple parsers are in here. but even that was kind of a lazy maneuver. i would support even moving those into their own files and dedicating this file to just the `Parser` class.

Reply:
I did a bit more restructuring than just this. The Parser ABC is now just called `Parser`  (in Parser.cs), and the subclasses are in separate files now. It feels a bit tidier that way.

I also moved the MapData/MarkData models out of the Parser namespace and into Models